### ISSUE TITLE
Export-Excel split arrays in one excel cell

### ISSUE THREAD

**Issue by DarkLite1** [dfinke/ImportExcel#43]
Hi Doug, I would like to say a big thank you again for your hard work on this function. I can't say it enough, it's an awesome function!

What I'm now trying to do is have an Excel sheet that contains multiple object properties in one cell.

Some code to clarify. This is the workflow I use to collect all the needed information about users home drives on servers:

``` powershell
Workflow Get-HomeDrivesHC {
    Param (
        [String[]]$ComputerName,
        [PSCredential]$Credentials,
        [Int]$OlderThanDays
    )

    Foreach -parallel ($C in $ComputerName) {
        Write-Verbose "Check ComputerName: $C"
        Sequence {
            $Drives = InlineScript {
                if (Test-Path -Path 'E:\HOME') {
                    Get-ChildItem -Path 'E:\HOME' -Force | where { $_.PSIsContainer }
                }
            } -PSComputerName $C -PSCredential $Credentials -PSConnectionRetryIntervalSec 3 -PSConnectionRetryCount 3

            Foreach -parallel ($D in $Drives) {
                Write-Verbose "Check drive: $C - $($D.FullName)"
                InlineScript {
                    Filter Select-FilesHC {
                        if ($_.LastWriteTime -lt (Get-Date).AddDays(-$Using:OlderThanDays)) {
                            Write-Output $_
                        }
                    }

                    $PathTooLong = $OldPST = $UsedPST = @()

                    $D = $Using:D
                    $Content  = Get-ChildItem -Path $D.FullName -Recurse -Force -EV +e -EA SilentlyContinue

                    $PSTfiles = $Content | where {$_.Extension -eq '.pst'}
                    $PSTsize  = $PSTfiles |  Measure-Object -Property length -sum -EA SilentlyContinue

                    foreach ($P in $PSTfiles) {
                        if ($P | Select-FilesHC) {
                            $OldPST += $P
                        }
                        else {
                            $UsedPST += $P
                        }
                    }

                    $FolderSize     = $Content | 
                                            Measure-Object -Property length -sum -EA SilentlyContinue
                    $TotalFileCount = $Content | 
                                            Measure-Object
                    $FilesPicture   = $Content | where {
                                        '.jpg', '.jpeg', '.png', '.bmp', '.jpg' -eq $_.Extension} | 
                                            Measure-Object -Property length -sum -EA SilentlyContinue
                    $FilesAudio     = $Content | where {
                                        '.mp3', '.wav', '.aac', '.ogg', '.wma', '.flac' -eq $_.Extension} | 
                                            Measure-Object -Property length -sum -EA SilentlyContinue
                    $FilesVideo     = $Content | where {
                                        '.mpeg', '.mpg', '.vob', '.flv', '.wmv', '.asf', 'mp4', 'm4v', '3gp' -eq $_.Extension} | 
                                            Measure-Object -Property length -sum -EA SilentlyContinue

                    $LastModTime    = $Content | Sort-Object LastWriteTime | Select -ExpandProperty LastWriteTime -Last 1

                    $D | Add-Member -MemberType noteproperty -Name FolderSize -Value $FolderSize.Sum
                    $D | Add-Member -MemberType noteproperty -Name TotalFileCount -Value $TotalFileCount.Count
                    $D | Add-Member -MemberType noteproperty -Name FilesPictureSize -Value $FilesPicture.Sum
                    $D | Add-Member -MemberType noteproperty -Name FilesPictureCount -Value $FilesPicture.Count
                    $D | Add-Member -MemberType noteproperty -Name FilesPSTsize -Value $PSTsize.Sum
                    $D | Add-Member -MemberType noteproperty -Name FilesPSTcount -Value $PSTsize.Count
                    $D | Add-Member -MemberType noteproperty -Name FilesAudioSize -Value $FilesAudio.Sum
                    $D | Add-Member -MemberType noteproperty -Name FilesAudioCount -Value $FilesAudio.Count
                    $D | Add-Member -MemberType noteproperty -Name FilesVideoSize -Value $FilesVideo.Sum
                    $D | Add-Member -MemberType noteproperty -Name FilesVideoCount -Value $FilesVideo.Count
                    $D | Add-Member -MemberType noteproperty -Name LastModTime -Value $LastModTime
                    $D | Add-Member -MemberType noteproperty -Name OldPST -Value $OldPST
                    $D | Add-Member -MemberType noteproperty -Name UsedPST -Value $UsedPST

                    $D

                    if ($e) {
                        # Report all paths that are too long
                        $e | where {$_.Exception -like 'System.IO.PathTooLongException*'} | foreach {
                            $PathTooLong += $_.TargetObject
                        }
                        New-Object PSObject -Property @{
                            PathTooLong = $PathTooLong
                        }
                    }
                } -PSComputerName $C -PSCredential $Credentials -PSConnectionRetryIntervalSec 3 -PSConnectionRetryCount 3
                Write-Verbose "Details retrieved for drive: $C - $($D.FullName)"
            }
            Write-Verbose "Details retrieved for computername: $C"
        }
    }
}

$Result = Get-HomeDrivesHC -ComputerName 'SERVER' -Credentials $Credentials -OlderThanDays '40'
```

This code works just fine and collects all the needed stuff in the variable $Result.

The desired result should look like this:

![image](https://cloud.githubusercontent.com/assets/10828570/10241741/c9cf6692-68eb-11e5-8141-977e45d53d8e.png)

So one line for multipe 'OldPSTFullName' files. I tried to do this as following:

``` powershell
$Result | Where Name | select Name, FullName, Exists, 
    @{N='OldPSTFullName';E={ if ($_.OldPst) { $_ | Select -ExpandProperty OldPst | Select -ExpandProperty FullName }}},
    @{N='OldPSTDate';E={ if ($_.OldPst) { $_ | Select -ExpandProperty OldPst | Select -ExpandProperty LastWriteTime }}} |
        Export-Excel -Path $ExcelFile -WorkSheetname 'HomeDrives' -TableName 'HomeDrives' -AutoSize -FreezeTopRow
```

The code above generates the text 'System.Object' in a cell. So I tried the following to avoid this:

``` powershell
$Result | Where Name | select Name, FullName, Exists, 
    @{N='OldPSTFullName';E={ if ($_.OldPst) { $_ | Select -ExpandProperty OldPst | Select -ExpandProperty FullName| Out-String}}},
    @{N='OldPSTDate';E={ if ($_.OldPst) { $_ | Select -ExpandProperty OldPst | Select -ExpandProperty LastWriteTime| Out-String}}} |
        Export-Excel -Path $ExcelFile -WorkSheetname 'HomeDrives' -TableName 'HomeDrives' -AutoSize -FreezeTopRow
```

The problem with the last one is that it's all on one row, and I would like to have it split. Maybe I'm doing something wrong on PowerShell level and it has less to do with your function. But I can't really find a way to figure this out.

Any help or tips would be greatly appreciated, thank you.

**Comment by Claustn** [comment]
Try creating an array of your objects that you want in the same cell, then join them with -join("`r`n")

**Comment by DarkLite1** [comment]
Thank you for the tip Claustin.

I tried the following code:

$Result | Where Name | select Name, FullName, Exists, 
    @{N='OldPSTFullName';E={ if ($_.OldPst) { ($_ | Select -ExpandProperty OldPst | Select -ExpandProperty FullName) -join("`r`n")}}},
    @{N='OldPSTDate';E={ if ($_.OldPst) { ($_ | Select -ExpandProperty OldPst | Select -ExpandProperty LastWriteTime) -join("`r`n")}}} |
        Export-Excel -Path $ExcelFile -WorkSheetname 'HomeDrives' -TableName 'HomeDrives' -AutoSize -FreezeTopRow

It looks like:
![image](https://cloud.githubusercontent.com/assets/10828570/10243008/b3cd1ac4-68f5-11e5-8c55-c18e9c9238f7.png)

When I click in the cell to rename it or something, it gets populated over multiple lines. But not by default... :(

**Comment by dfinke** [comment]
@DarkLite1 looking into it

@Claustn the \` here is interpreted as markdown. Escape backticks with a backslash \

\`r\`n

**Comment by dfinke** [comment]
Looks like you only need \`n. The challenge is, the `Style.WrapText` needs to be enabled. If `Style.WrapText` is enabled Excel will size the cells based on the content. This may not be desired result. 

Exposing `Style.WrapText` as a parameter could help solve this.

**Comment by dfinke** [comment]
I created a branch and added a -WrapText switch.

Excel decides where to wrap the text though. If you add a \`n to your data and then manually size the columns, the line feeds work. I don't like the row height adjustment when this is in place. 

It is a work in progress. 

https://github.com/dfinke/ImportExcel/tree/spikeTryWrapText

Please let me know what you think.
Then I will continue to tweak and see what makes the most sense.

**Comment by DarkLite1** [comment]
Good morning Doug, thank you for looking into this.

I've just tested your new code and it looks a bit weird indeed. 

These are the switches I used:
`Export-Excel -Path $ExcelFile -WorkSheetname 'HomeDrives' -TableName 'HomeDrives' -AutoSize -FreezeTopRow -WrapText`

This is the result:

![image](https://cloud.githubusercontent.com/assets/10828570/10275798/b45fa2d4-6b4a-11e5-8c38-8ca6bb197bf2.png)

I don't think we should add this, as it's not looking good at all. The strange thing is, that with your normal code, the wrapping only works when you click in wrapped box. Then the date divides itself properly over the cell. Is there not some kind of Excel refresh function or reapply formatting or something, that can be used to trigger this behavior by default?
